protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -551,22 +554,31 @@ func (rpccs *RPCConsumerServer) sendRelayToProvider(
 	usedProviders := relayProcessor.GetUsedProviders()
 	sessions, err := rpccs.consumerSessionManager.GetSessions(ctx, chainlib.GetComputeUnits(chainMessage), usedProviders, reqBlock, addon, extensions, chainlib.GetStateful(chainMessage), virtualEpoch)
 	if err != nil {
-		if lavasession.PairingListEmptyError.Is(err) && (addon != "" || len(extensions) > 0) {
-			err = utils.LavaFormatError("No Providers For Addon", err, utils.LogAttr("addon", addon), utils.LogAttr("extensions", extensions), utils.LogAttr("userIp", consumerIp))
+		if lavasession.PairingListEmptyError.Is(err) {
+			if addon != "" {
+				return utils.LavaFormatError("No Providers For Addon", err, utils.LogAttr("addon", addon), utils.LogAttr("extensions", extensions), utils.LogAttr("userIp", consumerIp))
+			} else if len(extensions) > 0 && relayProcessor.GetAllowSessionDegradation() { // if we have no providers for that extension, use a regular provider, otherwise return the extension results
+				sessions, err = rpccs.consumerSessionManager.GetSessions(ctx, chainlib.GetComputeUnits(chainMessage), usedProviders, reqBlock, addon, []*spectypes.Extension{}, chainlib.GetStateful(chainMessage), virtualEpoch)
+				if err != nil {
+					return err
+				}
+				relayProcessor.setSkipDataReliability(true) // disabling data reliability when disabling extensions.
+				relayRequestData.Extensions = []string{}
+			} else {
+				return err
+			}
+		} else {
+			return err
 		}
-		return err
+	}
+
+	// making sure next get sessions wont use regular providers
+	if len(extensions) > 0 {
```

Comment:
bug: this is triggered when the extensions fetch fails and we succeed bringing no extensions providers, you are resetting the relayRequestData.Extensions but not extensions